### PR TITLE
Add callstack context to cascade deletion debug logging

### PR DESCRIPTION
**Pull request by MrPetovan**
Part of #7134

See https://github.com/friendica/friendica/issues/7134#issuecomment-593795075

### CODE REVIEW

File: src/Worker/RemoveContact.php [hunk 1]
```diff
@@ -41,10 +41,10 @@ public static function execute($id) {
 		// Now we delete the contact and all depending tables
 		$condition = ['uid' => $contact['uid'], 'contact-id' => $id];
 		do {
-			$items = Item::select(['id'], $condition, ['limit' => 100]);
+			$items = Item::select(['id', 'guid'], $condition, ['limit' => 100]);
 			while ($item = Item::fetch($items)) {
 				Logger::notice('Delete removed contact item', ['id' => $item["id"]]);
-				DBA::delete('item', ['id' => $item['id']]);
+				DBA::delete('item', ['id' => $item['id'], 'guid' => $item['guid']]);
```
Comment (annando) [inline src/Worker/RemoveContact.php]: Why have you added the `guid` here? The `id` is the primary key, so adding the `guid` doesn't bring any benefit.

**Comment by annando** [comment]
Instead of something like `DBA::delete('item', ['id' => $item['id']]);` I would suggest `Item::delete('item', ['id' => $item['id']]);` - then the logging could be at a single place.

File: src/Worker/RemoveContact.php (hunk 1 above)
Comment (SpcCw) [inline src/Worker/RemoveContact.php]: This looks like a typo to me. I think you wanted to change the logger, not delete function?

**Comment by MrPetovan** [comment]
You both are correct.

**Comment by MrPetovan** [comment]
> Instead of something like `DBA::delete('item', ['id' => $item['id']]);` I would suggest `Item::delete('item', ['id' => $item['id']]);` - then the logging could be at a single place.

I agree, but it still wouldn't catch the cascade deletion. There are few enough cases that I was able to cover them all and we still couldn't catch the culprit, so centralizing the logging of the same calls won't change much in that regard.

**Comment by MrPetovan** [comment]
I edited the faulty commit and rebased the branch.

**Comment by annando** [comment]
Having only a single log call makes it easier having a look onto it since the test would always be the same. And when we are adding some more deletion we then don't have to add the logging there again.

One of my coding principles is that every time I'm doing similar stuff I'm putting it into a function that does the job.

**Comment by MrPetovan** [comment]
The problem with this centralized function is that we would have to require a full item array as a parameter, not for the deletion itself, but just for the logging call which I find icky. However it will be implied when we move the `Item` class to the domain-driven design model where we'll already have a fully hydrated item model anyway. However given the complexity it won't happen right away.

**Comment by MrPetovan** [comment]
Added callstack context to the second place we perform a dynamic `DELETE FROM` query.

File: src/Model/Item.php [hunk 2]
```diff
@@ -1106,30 +1106,30 @@ public static function deleteForUser($condition, $uid)
...
+		Logger::notice('Mark item for deletion by id', ['id' => $item_id, 'callstack' => System::callstack()]);
 		// locate item to be deleted
 		$fields = ['id', 'uri', 'uid', 'parent', 'parent-uri', 'origin',
 			'deleted', 'file', 'resource-id', 'event-id', 'attach',
 			'verb', 'object-type', 'object', 'target', 'contact-id',
 			'icid', 'iaid', 'psid'];
 		$item = self::selectFirst($fields, ['id' => $item_id]);
 		if (!DBA::isResult($item)) {
-			Logger::log('Item with ID ' . $item_id . " hasn't been found.", Logger::DEBUG);
+			Logger::debug('Item not found.', ['id' => $item_id]);
```
Comment (annando) [inline src/Model/Item.php]: The `DEBUG` log level corresponds with the `info` function.

**Comment by MrPetovan** [comment]
Thanks, I converted all the item deletion logging to `info`.

**Comment by annando** [comment]
Yeah, these log level constants are confusing.

**Comment by MrPetovan** [comment]
https://github.com/friendica/friendica-addons/pull/967 needs to be merged to ensure the pumpio addon still works.

**Pull request merged**